sky/engine/core/dart/window.dart

```diff
@@ -18,41 +23,102 @@ class WindowPadding {
   final double left;
 }
 
+/// An identifier used to select a user's language and formatting preferences,
+/// consisting of a language and a country. This is a subset of locale
+/// identifiers as defined by BCP 47.
 class Locale {
-  Locale(this.languageCode, this.countryCode);
+  const Locale(this.languageCode, this.countryCode);
 
+  /// The primary language subtag for the locale.
   final String languageCode;
+
+  /// The region subtag for the locale.
   final String countryCode;
 
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Locale)
+      return false;
+    final Locale typedOther = other;
+    return languageCode == typedOther.languageCode
+        && countryCode == typedOther.countryCode;
+  }
+
+  int get hashCode {
+    int result = 373;
+    result = 37 * result + languageCode.hashCode;
+    result = 37 * result + countryCode.hashCode;
+    return result;
+  }
+
   String toString() => '${languageCode}_$countryCode';
 }
 
+/// The most basic interface to the host operating system's user interface.
 class Window {
   Window._();
 
+  /// The number of device pixels for each logical pixel. This number might not
+  /// be a power of two. Indeed, it might not even be an integer. For example,
+  /// the Nexus 6 has a device pixel ratio of 3.5.
   double get devicePixelRatio => _devicePixelRatio;
   double _devicePixelRatio;
 
+  /// The dimensions of the rectangle into which the application will be drawn,
+  /// in logical pixels.
   Size get size => _size;
   Size _size;
 
+  /// The number of pixels on each side of the display rectangle into which the
+  /// application can render, but over which the operating system will likely
+  /// place system UI (such as the Android system notification area).
```

Comment:
Another good example is TVs which let you draw into an "overdraw" region that often isn't actually visible.